docs/conf.py

```diff
@@ -195,25 +186,185 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = "sphinx_rtd_theme"
-html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
+try:
+    import nvidia_sphinx_theme  # noqa: F401
+
+    html_theme = "nvidia_sphinx_theme"
+except ImportError:
+    import sphinx_rtd_theme
+
+    html_theme = "sphinx_rtd_theme"
+    html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
+
+
+html_theme_options = {
+    "switcher": {
+        # use for local testing
+        # "json_url": "http://localhost:8888/_static/switcher.json",
+        "json_url": "https://docs.nvidia.com/deeplearning/dali/user-guide/"
+        "docs/_static/switcher.json",
+        "version_match": "main" if "dev" in version_long else version_short,
+    },
+    "navbar_start": ["navbar-logo", "version-switcher"],
+    "primary_sidebar_end": [],
+}
+
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
 #
-html_theme_options = {
-    "canonical_url": "https://docs.nvidia.com/deeplearning/dali/user-guide/docs/index.html",
-    "collapse_navigation": False,
-    "display_version": True,
-    "logo_only": False,
-}
+html_theme_options.update(
+    {
+        "collapse_navigation": False,
+    }
+)
 
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
 html_static_path = ["_static"]
 
+
+switcher_path = os.path.join(html_static_path[0], "switcher.json")
+versions = []
+# the latest is in the archive
+for i in range(10, int(version_short.split(".")[1]) - 1):
+    if i >= 34:
+        versions.append((f"1.{i}", f"dali_1_{i}_0", "short_user"))
+    else:
+        versions.append((f"1.{i}", f"dali_1_{i}_0"))
+# add extra path version
+versions.append(("1.37.1", "dali_1_37_1", "short_user"))
+versions.append(("1.11.1", "dali_1_11_1"))
+# paths are different for 1.0-1.10
+for i in range(0, 10):
+    versions.append((f"1.{i}", f"dali_1{i}0"))
+# again different convention between 0.10 and 0.31
+for i in range(10, 30):
+    if i < 24:
+        if i <= 21:
+            versions.append((f"0.{i}", f"dali_0{i}0_beta", "devel"))
+        else:
+            versions.append((f"0.{i}", f"dali_0{i}0_beta"))
+    else:
+        versions.append((f"0.{i}", f"dali_0{i}0"))
+# add extra path version
+versions.append(("0.25.1", "dali_0251"))
+versions.append(("0.30", "dali_030"))
+versions.append(("0.31", "dali_031"))
+# and again different convention between 0.1 and 0.9
+for i in range(1, 10):
+    # no 0.0 release, we made 0.9.1
+    if i == 9:
+        continue
+    versions.append((f"0.{i}", f"dali_0{i}_beta", "devel"))
```

Comment:
This logic is quite convoluted and its only purpose is to produce a list of **past** releases, which is known to us and not that long. How about just pasting a sequence of `versions.append(...)` with those numbers instead of generating them? It will take more lines, but be much more readable than the loops

Reply:
> How about just pasting a sequence of versions.append(...) with those numbers instead of generating them?
I just don't want to manually update the list every time we release a new DALI. For sure I can trim it to a couple of most recent versions, but still, I would prefer to have it automatic.

Reply:
Of course we should keep the loops for generating entries for new releases, doing it manually would be a pain. But those are for old releases (0.1 to 0.9) if I read correctly, so we can just type them here once

Reply:
Much more readable for me, thanks!